kad/kad.go

```diff
@@ -28,3 +28,40 @@ type Key[K any] interface {
 	// and 0 if both keys are equal.
 	Compare(other K) int
 }
+
+// RoutingTable is the interface all Kademlia Routing Tables types support.
+type RoutingTable[K Key[K]] interface {
+	// Self returns the local node's Kademlia key
+	Self() K
+
+	// AddPeer tries to add a peer to the routing table
+	AddPeer(NodeID[K]) (bool, error)
```

Comment:
While I agree we should be consistent, I think having a mental model of `Nodes == Machines` is going to cause complications. We don't care about the physical arrangement.

A `Node` should be "something that participates in a Kademlia network" and "a node may have zero or more network addresses on which it can be contacted"

These things are called `Peers` in libp2p. A "libp2p Peer" is an instance of a "go-kademlia Node"

So, we should rename this method to `AddNode`

Reply:
So then let's use "Node" throughout `go-kademlia` 👍